src/Configuration.KeyPerFile/src/KeyPerFileConfigurationProvider.cs

```diff
@@ -73,6 +73,7 @@ private void Load(bool reload)
                 if (Source.Optional || reload) // Always optional on reload
                 {
                     Data = data;
+                    OnReload();
```

Comment:
Why is OnReload() not added before line 64?

Reply:
Thanks, that makes sense. Added the call to reload and a test to validate that scenario.